kafka/include/etc/confluent/docker/configure

```diff
@@ -16,35 +16,28 @@
 
 . /etc/confluent/docker/bash-config
 
-# If KAFKA_PROCESS_ROLES is defined it means we are running in KRaft mode
+# Ensure that KAFKA_PROCESS_ROLES and CLUSTER_ID are defined
+dub ensure KAFKA_PROCESS_ROLES
+dub ensure CLUSTER_ID
 
-# unset KAFKA_ADVERTISED_LISTENERS from ENV in KRaft mode when running as controller only
-if [[ -n "${KAFKA_PROCESS_ROLES-}" ]]
+# unset KAFKA_ADVERTISED_LISTENERS from ENV when running as controller only
+if [[ $KAFKA_PROCESS_ROLES == "controller" ]]
 then
-  echo "Running in KRaft mode..."
-  dub ensure CLUSTER_ID
-  if [[ $KAFKA_PROCESS_ROLES == "controller" ]]
+  if [[ -n "${KAFKA_ADVERTISED_LISTENERS-}" ]]
   then
-    if [[ -n "${KAFKA_ADVERTISED_LISTENERS-}" ]]
-    then
-      echo "KAFKA_ADVERTISED_LISTENERS is not supported on a KRaft controller."
-      exit 1
-    else
-      unset KAFKA_ADVERTISED_LISTENERS
-    fi
+    echo "KAFKA_ADVERTISED_LISTENERS is not supported on a KRaft controller."
+    exit 1
   else
-    dub ensure KAFKA_ADVERTISED_LISTENERS
-  fi 
+    unset KAFKA_ADVERTISED_LISTENERS
```

Comment:
if we've reached this case then doesn't that necessarily mean `KAFKA_ADVERTISED_LISTENERS` is not set?

Reply:
Yeah, I was being extra sure, but I think you're right, I can remove it